web/cobrands/peterborough/assets.js

```diff
@@ -163,42 +211,69 @@ fixmystreet.assets.add(light_defaults, {
     asset_item_message: ''
 });
 
-var bin_defaults = $.extend(true, {}, defaults, {
-    class: OpenLayers.Layer.PeterboroughVectorAsset,
-    select_action: true,
-    actions: {
-        asset_found: fixmystreet.message_controller.asset_found,
-        asset_not_found: fixmystreet.message_controller.asset_not_found
-    },
-    attributes: {
-        asset_details: function() {
-            var a = this.attributes;
-            return a.Reference + ", " + a.Location;
-        },
-        central_asset_id: 'OBJECTID'
-    },
-    asset_id_field: 'OBJECTID',
-    asset_type: 'spot'
-});
+var url_base = 'https://tilma.mysociety.org/resource-proxy/proxy.php?https://peterborough.assets/';
 
-fixmystreet.assets.add(bin_defaults, {
+var flytipping_defaults = $.extend(true, {}, arcgis_defaults, {
     http_options: {
       params: {
-        TYPENAME: 'LitterBins'
+        outFields: '',
       }
     },
-    asset_category: 'Litter bin',
-    asset_item: 'litter bin'
+    // this prevents issues when public and non public land
+    // are right next to each other
+    nearest_radius: 0.01,
+    stylemap: fixmystreet.assets.stylemap_invisible,
+    asset_category: waste_categories,
+    non_interactive: true,
+    road: true,
+    asset_item: 'road',
+    asset_type: 'road',
 });
 
-fixmystreet.assets.add(bin_defaults, {
+// PCC Property Combined
+fixmystreet.assets.add(flytipping_defaults, {
     http_options: {
-      params: {
-        TYPENAME: 'DogBins'
-      }
+      url: url_base + '2/query?',
     },
-    asset_category: 'Dog bin',
-    asset_item: 'dog waste bin'
+    actions: {
+        found: function(layer) {
+            $("#js-environment-message").addClass("hidden");
+        },
+        not_found: function() {
+            for ( var i = 0; i < fixmystreet.assets.layers.length; i++ ) {
+                var layer = fixmystreet.assets.layers[i];
+                if ( layer.fixmystreet.name == 'Adopted Highways' && layer.selected_feature ) {
+                    $("#js-roads-responsibility").addClass("hidden");
+                    return;
+                }
+            }
+            $("#js-roads-responsibility").addClass("hidden");
+            $('#js-environment-message').removeClass('hidden');
+            fixmystreet.message_controller.check_for_stopper();
+        },
+    }
+});
+
+// PCC Property Leased Out NOT Responsible
+fixmystreet.assets.add(flytipping_defaults, {
+    http_options: {
+      url: url_base + '3/query?',
+    },
+    actions: {
+        found: function() {
+            $("#js-roads-responsibility").addClass("hidden");
+            $('#js-environment-message').removeClass('hidden');
+        },
+        not_found: function() {
```

Comment:
Is this function needed? Seems like the other layer would do what's wanted here, and hide the message on a road/show otherwise, this function is just repeating the same thing. I might have missed something. A test would make it easy to tell :)